Serialization/ArgsList.Serialization.cs

```diff
@@ -172,7 +225,7 @@ public static SupportedType Create<T>(Func<ArgsList, T[]> getList, Action<ArgsLi
 			{
 				return new SupportedType(
 					typeof(T),
-					i => getList(i).Cast<object>().ToList(),
+					i => (getList(i) ?? new T[0]).Cast<object>().ToList(),
```

Comment:
If it's null why make an array, cast and then make the list? Could you just new the list straight away?

Reply:
It was less code, each line is a one line function otherwise it looks like this

```
				return new SupportedType(
					typeof(T),
					i =>
					{
						var list = getList(i);
						return list != null ? list.Cast<object>().ToList() : new List<object>();
					},
					(i, v) => setList(i, v.Cast<T>().ToArray()));
```


true enough though, this is more optimal